src/pausing/storage.rs

```diff
@@ -0,0 +1,113 @@
+use crate::events::ResponseHandler;
+use crate::pausing::contract_paused_error;
+use crate::pausing::events::{ContractPausedEvent, ContractResumedEvent};
+use cosmwasm_std::{Env, StdResult, Storage};
+use cw_storage_plus::Item;
+
+const PAUSED_SINCE_BLOCK: Item<u64> = Item::new("paused_since");
+
+pub fn is_paused_since_block(storage: &dyn Storage) -> StdResult<Option<u64>> {
+    PAUSED_SINCE_BLOCK.may_load(storage)
+}
+
+pub fn pause_contract(
+    storage: &mut dyn Storage,
+    response_handler: &mut ResponseHandler,
+    since_block: u64,
+) -> StdResult<()> {
+    response_handler.add_event(ContractPausedEvent {
+        since_block: &since_block,
+    });
+    PAUSED_SINCE_BLOCK.save(storage, &since_block)
+}
```

Comment:
Smething tells me we, that it would be betetr if the `since_block` would be optional = would have the `Option<u64>` type.

That would allow to use this single method to pause and upause using the idiom od deletions - something like below:
```suggestion
pub fn pause_contract(
    storage: &mut dyn Storage,
    response_handler: &mut ResponseHandler,
    since_block: &Option<u64>,
) -> StdResult<()> {
    if let Some(paused_since_block) = since_block {
        response_handler.add_event(ContractPausedEvent {
            since_block: &since_block,
        });
        PAUSED_SINCE_BLOCK.save(storage, &since_block);
    } else {
        resume_contract(storage, respnse_handler);
    }
}
```

Reply:
Never mind above, I think it is fine as it is - it is simpler.